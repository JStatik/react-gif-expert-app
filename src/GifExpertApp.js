import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {
    const [ categorias, setCategorias ] = useState( [ 'Vikingos' ] );

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory setCategorias={ setCategorias } />
            <hr/>

            <ol>
                {
                    categorias.map( ( elem ) => {
                        return <GifGrid key={ elem } categoria={ elem } />
                    } ) 
                }
            </ol>
        </>
    )
};

export default GifExpertApp;
