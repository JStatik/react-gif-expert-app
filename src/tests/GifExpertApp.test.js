import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';

describe( 'Pruebas en el componente GifExpertApp', () => {
    let wrapper = shallow( <GifExpertApp /> );

    beforeEach( () => {
        wrapper = shallow( <GifExpertApp /> );
    } );

    test( 'Debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    } );

    test( 'Debe tener un componente GifGrid', () => {
        expect( 1 ).toBe( wrapper.find( 'GifGrid' ).length );
    } );
} );
