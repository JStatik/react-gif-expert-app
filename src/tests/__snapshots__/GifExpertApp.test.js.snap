// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Vikingos"
      key="Vikingos"
    />
  </ol>
</Fragment>
`;
